### ISSUE TITLE
You accidentially made it just a token logger that doesn't snipe nitro links

### ISSUE THREAD

**Issue by dreggsome** [Nexons/bdnitrosniper#3]

**Comment by LarchyEverything** [comment]
Such a shame.

**Comment by Ardougne** [comment]
I looked at the code and can't find anything that would make you think it's a token logger. If you could reply with the part that is a token logger that would be great.

**Comment by MaiSan9999** [comment]
> 

It's just getting the token to check the validity

**Comment by cooleybyte** [comment]
dont fanboy maisan. the fact is this doesnt even work even IF you install it. shame on you for defending this crap

**Comment by MaiSan9999** [comment]
> dont fanboy maisan. the fact is this doesnt even work even IF you install it. shame on you for defending this crap

I never used the plugin in fact i had powercord and never installed bdcompact  afaik (my sma knowledge) it doesn't post any token to webhook i never said to trust the guy and give bank account to him or shit like that

**Comment by Ardougne** [comment]
stop being poor just buy nitro

**Comment by cooleybyte** [comment]
Liar. It doesn't work

**Comment by Woosh9129** [comment]
> > dont fanboy maisan. the fact is this doesnt even work even IF you install it. shame on you for defending this crap
> 
> I never used the plugin in fact i had powercord and never installed bdcompact afaik (my sma knowledge) it doesn't post any token to webhook i never said to trust the guy and give bank account to him or shit like that

as a coder, i agree. this does not post your token

**Comment by AwesomePotatoCodes** [comment]
> > > dont fanboy maisan. the fact is this doesnt even work even IF you install it. shame on you for defending this crap
> > 
> > 
> > I never used the plugin in fact i had powercord and never installed bdcompact afaik (my sma knowledge) it doesn't post any token to webhook i never said to trust the guy and give bank account to him or shit like that
> 
> as a coder, i agree. this does not post your token

true! the only POST request made using the token is to actually validate a session on the discord api. other than that, there's no POST requests made to third-party shady websites that are token grabbers.

it's safe, but the odds of this working are somewhat low from what i understand?

**Comment by cooleybyte** [comment]
How bout kill yourself bitch

**Comment by Ardougne** [comment]
lol

**Comment by cooleybyte** [comment]
no really. die in a fire

**Comment by cooleybyte** [comment]
ie irl and choke. kill yourself
dickhead

**Comment by DarkTurns14** [comment]
Not entirely correct, the code does not directly log your Discord token to any external source. However, it does store the Discord tokens temporarily in the tokens array while searching for a valid token to use for authentication with Discord's API. The tokens are retrieved from the user's local storage directory where Discord stores authentication tokens. Once a valid token is found and used for authentication, it is not logged anywhere in the code provided.

It defines a class named NitroSniper and exports it as a module.
The class has methods to retrieve information like name, description, version, and author of the bot.
The start() method is an asynchronous function that initializes the bot.
It reads Discord tokens from the user's Discord local storage.
It iterates over the files in the local storage directory to extract tokens.
It sends requests to Discord's API to check if the tokens are valid.
If a valid token is found, it logs in the bot and patches the dispatch function to listen for new messages.
The dispatch() function is called whenever a new message is received.
It checks if the message contains a Discord nitro gift code.
If a gift code is found, it attempts to redeem it using the stored token.
Depending on the response from Discord's API, it displays success or error messages.

In conclusion this is not a token logger, its a poorly made nitro sniper.
